network-store-integration-test/src/test/java/com/powsybl/network/store/integration/NetworkStoreIT.java

```diff
@@ -1295,6 +1304,126 @@ public void internalConnectionsFromCgmesTest() {
+            assertTrue(cgmesSvMetadata.getDescription().contains("CGMES Conformity Assessment"));
+            assertEquals(4, cgmesSvMetadata.getSvVersion());
+            assertEquals("http://A1.de/Planning/ENTSOE/2", cgmesSvMetadata.getModelingAuthoritySet());
+            assertEquals(3, cgmesSvMetadata.getDependencies().size());
+            assertEquals(CgmesTopologyKind.NODE_BREAKER, cimCharacteristics.getTopologyKind());
+            assertEquals(16, cimCharacteristics.getCimVersion());
+
+            CgmesSvMetadataAttributes cgmesSvMetadataAttributes = CgmesSvMetadataAttributes.builder()
+                    .description("Description")
+                    .svVersion(6)
+                    .dependencies(new ArrayList<>())
+                    .modelingAuthoritySet("modelingAuthoritySet")
+                    .build();
+
+            ((NetworkImpl) readNetwork).getResource().getAttributes().setCgmesSvMetadata(cgmesSvMetadataAttributes);
+
+            CimCharacteristicsAttributes cimCharacteristicsAttributes = CimCharacteristicsAttributes.builder()
+                    .cimVersion(5)
+                    .cgmesTopologyKind(CgmesTopologyKind.BUS_BRANCH)
+                    .build();
+
+            ((NetworkImpl) readNetwork).getResource().getAttributes().setCimCharacteristics(cimCharacteristicsAttributes);
+
+            service.flush(readNetwork);
+        }
+
+        try (NetworkStoreService service = createNetworkStoreService()) {
+
+            Map<UUID, String> networkIds = service.getNetworkIds();
+
+            assertEquals(1, networkIds.size());
+
+            Network readNetwork = service.getNetwork(networkIds.keySet().stream().findFirst().get());
+
+            CgmesSvMetadata cgmesSvMetadata = readNetwork.getExtensionByName("cgmesSvMetadata");
+            CimCharacteristics cimCharacteristics = readNetwork.getExtensionByName("cimCharacteristics");
+
+            assertEquals(CgmesTopologyKind.BUS_BRANCH, cimCharacteristics.getTopologyKind());
+            assertEquals(5, cimCharacteristics.getCimVersion());
+            assertEquals("Description", cgmesSvMetadata.getDescription());
+            assertEquals(6, cgmesSvMetadata.getSvVersion());
+            assertEquals("modelingAuthoritySet", cgmesSvMetadata.getModelingAuthoritySet());
+            assertEquals(0, cgmesSvMetadata.getDependencies().size());
+
+            cgmesSvMetadata = new CgmesSvMetadataImpl((NetworkImpl) readNetwork,
+                    "Description2",
+                    7,
+                    new ArrayList<>(),
+                    "modelingAuthoritySet2");
+
+            cimCharacteristics = new CimCharacteristicsImpl((NetworkImpl) readNetwork,
+                    CgmesTopologyKind.NODE_BREAKER,
+                    6);
+
+            readNetwork.addExtension(CgmesSvMetadata.class, cgmesSvMetadata);
+            readNetwork.addExtension(CimCharacteristics.class, cimCharacteristics);
```

Comment:
Maybe add tests here to verify that CgmesSvMetadata and CimCharacteristics attributes can be modified one by one (i.e. using the CgmesSvMetadata and CimCharacteristics attributes setters) and that the resulting extensions are correctly persisted).

Reply:
Not to be done as setters are not required.